javaslang/src/main/java/javaslang/collection/SortedMap.java

```diff
@@ -71,8 +71,10 @@
     /**
      * Returns the underlying key-comparator which defines the order of the elements contained in this map.
      *
+     * @deprecated Use {@link SortedMap#comparator()} instead
      * @return This map's key-comparator.
      */
+    @Deprecated
```

Comment:
Do you think it is better to carry our deprecated methods into the next major version? We have several breaking API changes. Removing all unnecessary things makes Javaslang simpler. However, we should provide a migration plan to our users.